modules/cli/src/main/scala/scala/cli/commands/package0/Package.scala

```diff
@@ -85,6 +89,19 @@ object Package extends ScalaCommand[PackageOptions] with BuildCommandHelpers {
         configDb.get(Keys.actions).getOrElse(None)
       )
 
+    def detectNativePackageType(
+      platform: Platform,
+      sn: ScalaNativeOptions
+    ): Option[Native & scala.deriving.Mirror.Singleton] =
```

Comment:
Can you explain how `scala.deriving.Mirror.Singleton` works? Do we need to use it here?